Dockerfile

```diff
@@ -26,8 +27,7 @@ COPY pkg/ pkg/
 ADD $ODH_MANIFESTS_URL $LOCAL_BUNDLE
 RUN echo "$ODH_MANIFESTS_REF" > MANIFEST_VERSION && chmod g+r $LOCAL_BUNDLE
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
-
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} go build -a -o manager main.go
```

Comment:
isnt this forcing to pass in new argument when calling build?  it will break anyone who is not setting this variable. 
should default to `amd64`

Reply:
We don't need to pass any new argument.
If we call make docker-build in a local env which has the Apple Silicon M1 SO
the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore, by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
Ref:
https://docs.docker.com/engine/reference/builder/?_gl=1*mdpnbe*_ga*NTcyMTk5MDI1LjE2NjE1MDIyOTA.*_ga_XJWPQMJYHQ*MTY5MTEyNjg4Ni45MS4xLjE2OTExMjcwNTUuNjAuMC4w#automatic-platform-args-in-the-global-scope
Also same change is present in https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/Makefile

Reply:
I've tested it with Podman and it sets these 2 env variables on the fly, it should work